client/data/sites/use-site-excerpts-query.ts

```diff
@@ -41,6 +41,9 @@ export const useSiteExcerptsQuery = () => {
 			sites: [],
 		},
 	} );
+
+	// usQuery isLoading is false, so we need to check isFetching for the first time
+	return { ...result, isLoading: result.isFetching && ! result.data?.length };
```

Comment:
[Reading the docs for `isLoading`](https://react-query-v3.tanstack.com/reference/useQuery) it seems like it's exactly what we want, and I think it's best to leave the meaning for these standard ReactQuery objects as-is so that other devs who are familiar with ReactQuery know how they work.

I think the issue is the `placeholderData` option we pass to `useQuery`. Since we're providing a placeholder it means the query is never in a "hard" loading state, and so `isLoading` is always false. Now that you're implementing a proper placeholder UI I think we should remove `placeholderData` option and leave the `isLoading` flag as-is.

Reply:
@p-jackson Thank you!! , Yesterday I tried to remove the `initialData` to solve the isLoading issue, but I didn't try removing the `placeholderData`.

Now the default `isLoading` is working as expected 💪